### PR TITLE
NixOS/zabbix-{server,agent}: 1.8 -> 3.4 wip

### PR DESCRIPTION
**Pull request by srhb**
###### Motivation for this change
Currently, the zabbix modules use a very old version of zabbix. Most of the mechanics appear to have been unaltered, but there are some fixes necessary to bring it to work with a current version. I'm unsure if this module is really used by anyone (because of its age.) If it isn't, I would like to rip it apart entirely and parameterize, among other things, the actual package version used by the module.

I notice that @edolstra is marked as maintainer, so I'm curious if you have strong feelings on keeping this module as-is, or whether I should take a less conservative approach than this on bringing it up-to-date.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by edolstra** [comment]
Feel free to take over as maintainer for Zabbix. I'm not using it anymore at the moment.

File: pkgs/servers/monitoring/zabbix/1.8.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{ stdenv, fetchurl, pkgconfig, postgresql, curl, openssl, zlib }:
+
+let
+
+  version = "1.8.22";
```
Comment (Mic92) [inline pkgs/servers/monitoring/zabbix/1.8.nix]: I don't think we need both zabbix versions. It also seems the old versions would not be maintained well.

**Comment by srhb** [comment]
Thank you both for the comments. I will take over as maintainer, remove the old versions and revamp the modules in the next week or so. :)

**Comment by adamtulinius** [comment]
This might be interested when figuring out which versions to keep, and which to get rid of: https://www.zabbix.com/life_cycle_and_release_policy

But if nobody has strong opinions on this, it's probably wasted to spend any effort on Zabbix 1.x and 2.x.

**Comment by Mic92** [comment]
we have merge conflicts here.

**Comment by srhb** [comment]
OK, looks like I'll have to revive 2.0 and 2.2 for now...

**Comment by srhb** [comment]
Wait, no, 2.0 is EOL. I'll revive 2.2 only and try and see if the modules still work.

**Comment by yesbox** [comment]
The Zabbix web service (nixos/modules/services/web-servers/apache-httpd/zabbix.nix) is currently broken. It seems the old Zabbix server isn't PHP 7 compatible and so it doesn't serve anything. Maybe for the better, I would be very surprised if it isn't full of known vulnerabilities.

> [php7:error] PHP Fatal error:  'break' not in the 'loop' or 'switch' context in /nix/store/hzr8p3xifm4912f3h89qppkcp61kdrgw-zabbix-1.8.22/share/zabbix/php/include/maps.inc.php on line 126

**Comment by srhb** [comment]
I also no longer have to use Zabbix, closing.

**Comment by Mic92** [comment]
In the current state we probably should remove zabbix until someone maintains it properly.

**Comment by adamtulinius** [comment]
I've considered fixing it multiple times, but lost the will just by thinking about all the old cruft that had to be supported (v 1.8, 2.0, 2.2). If it was an acceptable solution to *only* support zabbix 3.4, then the task would be much easier.

**Comment by srhb** [comment]
@adamtulinius Only supporting the newest is, at it stands, way more reasonable than having every version be unsupported, so please don't let that perceived requirement hold you back. :)

**Pull request closed without merging**